packages/web3-ccip-read/src/index.js

```diff
@@ -0,0 +1,167 @@
+/*
+ This file is part of web3.js.
+
+ web3.js is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Lesser General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ web3.js is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file index.js
+ * @author Leon Talbert <leon@ens.domains>
+ * @date 2022
+ */
+
+var abi = require('web3-eth-abi');
+var http = require('web3-http');
+var {defaultAbiCoder, Interface} = require('@ethersproject/abi');
+var {hexConcat} = require('@ethersproject/bytes');
+
+const ENCODED_CCIP_READ_FUNCTION_SELECTOR = '556f1830';
+const MAX_REDIRECT_COUNT = 3;
+const OFFCHAIN_LOOKUP_PARAMETER_TYPES = ['address', 'string[]', 'bytes', 'bytes4', 'bytes'];
+
+const CCIP_READ_INTERFACE = new Interface([
+    'function callback(bytes memory result, bytes memory extraData)',
+]);
+
+
+var _callGateway = async function (urls, to, callData) {
+    var httpObject = new http.Http();
```

Comment:
The specification suggest: 

> The URLs returned by a contract may be of any schema, but this specification only defines how clients should handle HTTPS URLs.

So there should be a pluggable approach to handle further protocols for gateway.

Reply:
Thanks you are right, this has been implemented now